File: Parallel and Distributed Programming/Laborator2/Lab2Java/src/multiplication/DigitMultiplication.java
package multiplication;

import java.util.ArrayList;
import java.util.List;

/**
 * Created on: 29-Oct-19, 00:13
 *
 * @author: Katy Bucșa
 */

public class DigitMultiplication {
    public static List<Integer> multiplyVector(List<Integer> vector, int number) {
        List<Integer> result = new ArrayList<>();
        int forA = 0, s;
        for (int i = vector.size() - 1; i >= 0; i--) {
            s = vector.get(i) * number + forA;
            result.add(0,s % 10);
            forA = s / 10;
        }
        if (forA != 0) {
            result.add(0,forA);
        }
        return result;
    }
}

File: Parallel and Distributed Programming/Laborator2/Lab2C++/Lab2C++/DigitMultiplication.cpp
#include <vector>

static class DigitMultiplication {
public:
	static std::vector<int> multiplyVector(std::vector<int>& vector, int number) {
		std::vector<int> result;
		int forA = 0, s;
		for (int i = vector.size() - 1; i >= 0; i--) {
			s = vector.at(i) * number + forA;
			result.insert(result.begin(), s % 10);
			forA = s / 10;
		}
		if (forA != 0) {
			result.insert(result.begin(), forA);
		}
		return result;
	}
};